ios/Classes/AblyFlutterPlugin.m

```diff
@@ -39,6 +39,44 @@ -(nullable AblyFlutter *)ablyWithHandle:(NSNumber *)handle;
     [plugin registerWithCompletionHandler:result];
 };
 
+static FlutterHandler _createRestWithOptions = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
+    AblyFlutterMessage *const message = call.arguments;
+    LOG(@"message for handle %@", message.handle);
+    AblyFlutter *const ably = [plugin ablyWithHandle:message.handle];
+    // TODO if ably is nil here then an error response, perhaps? or allow Dart side to understand null response?
+    result([ably createRestWithOptions:message.message]);
+};
+
+static FlutterHandler _publishRestMessage = ^void(AblyFlutterPlugin *const plugin, FlutterMethodCall *const call, const FlutterResult result) {
```

Comment:
@QuintinWillison I may need some enlightenment here, but could do these methods not directly map to actual IDL methods instead of `publishRestMessage` which does not exist in our IDL.  Is there a reason for this that I am not aware of due to the constraints of Dart?  I do understand there are issues with overloaded methods with Dart, but I am surprised to see code like this where, as the reviewer, I am not entirely clear what I am actually reviewing.

@tiholic separately, could I ask that we include code examples in the README for methods we are delivering as we go along?  If you look at other SDKs, such as Ruby and the publish method, the high level methods are documented in the README - https://github.com/ably/ably-ruby#publishing-a-message-to-a-channel-1.  We don't need to document every method (which is what we have done in all READMEs), however it is imperative for me to see what API interface we are exposing to delivers when using the Flutter SDK.  Could you aim to do that as you go along?

Reply:
@mattheworiordan ...

The particular block name you've picked up on (`_publishRestMessage`) is an internal implementation detail unique to the Objective-C code (i.e. just a third of the solution, and in this case just this file - it's `static`!). Right now I feel we need to concentrate on the public API which basically means what's presented to Flutter users in pure Dart code. Any illogical mappings will naturally flat to the surface as that evolves.

Regarding your comment to @tiholic relating to examples... The Flutter-plugin-idomatic way of delivering example usage, as I understand it, is in the code in the example folder. It's created for you, out of the box, when you use Flutter's tooling to [create a plugin project](https://flutter.dev/docs/development/packages-and-plugins/developing-packages#step-1-create-the-package-1). It may make sense to also have examples in the readme, or perhaps an indirection from there to the 'live and runnable' Flutter example code. This may need more thought but I think that will shake out a bit more once we have enough implementation in place to do something useful with the plugin!

Reply:
Ok, thanks for explaining.  

> Right now I feel we need to concentrate on the public API which basically means what's presented to Flutter users in pure Dart code.

Agreed

> The Flutter-plugin-idomatic way of delivering example usage, as I understand it, is in the code in the example folder.

What wasn't what I meant.  What I meant is is the Ably API we provide from the Flutter SDK idiomatic to Dart developers?  If not, what are the trade offs?  

I was simply requesting that as we go along, we include the code samples in the README to be consistent with our SDKs - hence https://github.com/ably/ably-ruby#publishing-a-message-to-a-channel-1

Reply:
@mattheworiordan @QuintinWillison 

will have the high level API available in README in next PR.